src/Dialogs/Weather/EdlControlsWidget.cpp

```
@@ -0,0 +1,279 @@
+}
+
+static void
+ApplyEdlLevelAuto() noexcept
+{
+  EDL::UpdateCurrentLevel();
+}
+
+void
+EdlControlsWidget::HandleWeatherOverlayInput(const char *misc) noexcept
+{
+  switch (WeatherMapOverlay::ParseOverlayInputAction(misc)) {
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)